Sources/SwiftProtobuf/DelimitedMessageUtil.swift

```diff
@@ -0,0 +1,215 @@
+// Sources/SwiftProtobuf/DelimitedMessageUtil.swift - Delimited support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Helpers to read/write message with a length prefix.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Additional errors for delimited message handing.
+public enum DelimitedMessageError: Error {
+  /// If a read/write to the stream fails, but the stream's `streamError` is nil,
+  /// this error will be throw instead since the stream didn't provide anything
+  /// more specific. A common cause for this can be failing to open the stream
+  /// before trying to read/write to it.
+  case unknownStreamError
+
+  /// While reading/writing to the stream, less than the expected bytes was
+  /// read/written.
+  case truncated
+}
+
+/// Helper methods for reading/writing messages with a length prefix.
+public enum DelimitedMessageUtil {
```

Comment:
Raising up from the now "outdated comment" I added this too:

We could also capture better that is specific to the binary serialization format with something like `DelimitedBinarySerialization` (or something like that).  And that would tie things together instead of trying to get those names into the function names themselves.